pkg/cluster/config/v1alpha2/conversion.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by conversion-gen. DO NOT EDIT.
+
+package v1alpha2
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	config "sigs.k8s.io/kind/pkg/cluster/config"
+)
+
+func Convert_v1alpha2_Config_To_config_Cluster(in *Config, out *config.Cluster, s conversion.Scope) error {
+	for _, node := range in.Nodes {
+		// TODO(bentheelder): if v1alpha3 does re-gain per-node patches, drop
+		// this snippet
+		// Copy config patches from nodes to the cluster level, since we only
+		// generate a cluster-wide config anyhow
+		for _, patch := range node.KubeadmConfigPatches {
+			out.KubeadmConfigPatches = append(out.KubeadmConfigPatches, patch)
+		}
+		for _, patch := range node.KubeadmConfigPatchesJSON6902 {
+			out.KubeadmConfigPatchesJSON6902 = append(out.KubeadmConfigPatchesJSON6902, patch)
+		}
+
+		// skip now-implicit external load balancers
+		if node.Role == ExternalLoadBalancerRole {
+			continue
+		}
+
+		// always convert the node
+		convertedNode := config.Node{}
+		if err := Convert_v1alpha2_Node_To_config_Node(&node, &convertedNode, s); err != nil {
+			return err
+		}
+		out.Nodes = append(out.Nodes, convertedNode)
```

Comment:
What if someone specified `replicas: 0`?

Reply:
What a jerk /s
We should test how 0.1.0 behaves specifically and reflect that.

Reply:
i did test it at some point and i think it was silently existing. :)

Reply:
```
$ kind version
0.1.0

$ cat $HOME/kind-config.yaml
kind: Config
apiVersion: kind.sigs.k8s.io/v1alpha2
nodes:
- role: control-plane
  replicas: 0

$ kind create cluster --config=$HOME/kind-config.yaml
Error: failed to create cluster: please add at least one node with role "control-plane"

Usage:
  kind create cluster [flags]

Flags:
      --config string   path to a kind config file
  -h, --help            help for cluster
      --image string    node docker image to use for booting the cluster
      --name string     cluster context name (default "1")
      --retain          retain nodes for debugging when cluster creation fails
      --wait duration   Wait for control plane node to be ready (default 0s)

Global Flags:
      --loglevel string   logrus log level [panic, fatal, error, warning, info, debug] (default "warning")

failed to create cluster: please add at least one node with role "control-plane"
```

Reply:
so we should actually treat it as 0 and skip the node I guess.

Reply:
will handle this in a follow-up once we decide